plonky2/src/plonk/prover.rs

```diff
@@ -36,6 +36,13 @@ where
     [(); C::Hasher::HASH_SIZE]:,
 {
     let config = &common_data.config;
+
+    #[cfg(not(feature = "rand"))]
```

Comment:
Likewise here; seems like this shouldn't be reachable since you added the assert in `check_config`.